src/ansi/iterator.rs

```diff
@@ -162,139 +165,111 @@ impl vte::Perform for Performer {
     }
 }
 
-// Based on https://github.com/alacritty/alacritty/blob/57c4ac9145a20fb1ae9a21102503458d3da06c7b/alacritty_terminal/src/ansi.rs#L1168
-fn ansi_term_style_from_sgr_parameters(parameters: &[i64]) -> ansi_term::Style {
-    let mut i = 0;
+// Based on https://github.com/alacritty/alacritty/blob/9e71002e40d5487c6fa2571a3a3c4f5c8f679334/alacritty_terminal/src/ansi.rs#L1175
+fn ansi_term_style_from_sgr_parameters(params: &mut ParamsIter<'_>) -> ansi_term::Style {
     let mut style = ansi_term::Style::new();
-    loop {
-        if i >= parameters.len() {
-            break;
-        }
+    while let Some(param) = params.next() {
+        match param {
+            // [0] => Some(Attr::Reset),
+            [1] => style.is_bold = true,
+            [2] => style.is_dimmed = true,
+            [3] => style.is_italic = true,
+            [4] => style.is_underline = true,
```

Comment:
Technically this should probably be as follows:

```suggestion
            [4, ..] => style.is_underline = true,
```

Since it's possible to have underlines with more than one parameter separated by colons.